File: OOP-and-Functional-Programming_Samples/OOP/Simulated_Assembly_instructions_and_Samples/c++/add.cc
#include "add.h"
using namespace std;

void Add::ejecutar(Programa* programa){
    stack<int>& pila = programa->getPila();
    int a = pila.top();
    pila.pop();
    int b = pila.top();
    pila.pop();

    pila.push(a+b);

    programa->inc();
}

string Add::mostrar(){
    return "add";
}

File: OOP-and-Functional-Programming_Samples/OOP/Simulated_Assembly_instructions_and_Samples/java/Add.java
import java.util.Stack;

public class Add extends Instruccion{

    public void ejecutar(Programa programa){
        Stack<Integer> pila = programa.getPila();
        int a = pila.pop();
        int b = pila.pop();
        pila.push(a + b);
        programa.inc();
    }

    public String mostrar(){
        return "add";
    }
}
